axelor-budget/src/main/java/com/axelor/apps/budget/service/GlobalBudgetService.java

```diff
@@ -37,4 +38,14 @@ GlobalBudget generateGlobalBudgetWithTemplate(GlobalBudgetTemplate globalBudgetT
   GlobalBudget changeBudgetVersion(
       GlobalBudget globalBudget, BudgetVersion budgetVersion, boolean needRecomputeBudgetLine)
       throws AxelorException;
+
+  void updateGlobalBudgetDates(GlobalBudget globalBudget) throws AxelorException;
+
+  void generateBudgetKey(GlobalBudget globalBudget) throws AxelorException;
+
+  List<Budget> getAllBudgets(GlobalBudget globalBudget);
+
+  List<Long> getAllBudgetIds(GlobalBudget globalBudget);
+
+  void fillGlobalBudgetOnBudget(GlobalBudget globalBudget);
```

Comment:
should be moved either into a separate service, or a protected method in GlobalBudgetManagementRepository since it looks like it is only called here.